src/simplewallet/simplewallet.cpp

```diff
@@ -3980,51 +4126,270 @@ void simple_wallet::stop()
-  if (args.size() == 0)
+  auto local_args = args;
+  if(!local_args.empty() && tools::wallet2::parse_short_payment_id(local_args.back(), payment_id))
+  {
+    local_args.pop_back();
+  }
+  else
   {
     payment_id = crypto::rand<crypto::hash8>();
-    success_msg_writer() << tr("Random payment ID: ") << payment_id;
-    success_msg_writer() << tr("Matching integrated address: ") << m_wallet->get_account().get_public_integrated_address_str(payment_id, m_wallet->testnet());
-    return true;
   }
-  if(tools::wallet2::parse_short_payment_id(args.back(), payment_id))
+  if (local_args.size() > 1)
   {
-    success_msg_writer() << m_wallet->get_account().get_public_integrated_address_str(payment_id, m_wallet->testnet());
+    fail_msg_writer() << tr("failed to parse payment id");
     return true;
   }
-  else {
-    bool has_payment_id;
-    crypto::hash8 payment_id;
-    account_public_address addr;
-    if(get_account_integrated_address_from_str(addr, has_payment_id, payment_id, m_wallet->testnet(), args.back()))
+
+  uint32_t index_minor = 0;
+  if (local_args.size() == 1)
+  {
+    if(!epee::string_tools::get_xtype_from_string(index_minor, local_args[0]))
     {
-      if (has_payment_id)
-      {
-        success_msg_writer() << boost::format(tr("Integrated address: account %s, payment ID %s")) %
-          get_account_address_as_str(m_wallet->testnet(),addr) % epee::string_tools::pod_to_hex(payment_id);
-      }
-      else
-      {
-        success_msg_writer() << tr("Standard address: ") << get_account_address_as_str(m_wallet->testnet(),addr);
-      }
+      fail_msg_writer() << tr("failed to parse index");
       return true;
     }
   }
-  fail_msg_writer() << tr("failed to parse payment ID or address");
+
+  if (index_minor >= m_wallet->get_num_subaddresses())
+  {
+    fail_msg_writer() << tr("specify an index between 0 and ") << (m_wallet->get_num_subaddresses() - 1) << tr(", or use \"address new\"");
+    return true;
+  }
+
+  success_msg_writer() << m_wallet->get_integrated_subaddress_as_str(index_minor, payment_id);
+  success_msg_writer() << tr("Integrated payment ID: ") << epee::string_tools::pod_to_hex(payment_id);
+  success_msg_writer() << tr("Address index: ") << index_minor;
+  return true;
+}
+//----------------------------------------------------------------------------------------------------
+bool simple_wallet::parse_address(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  if (args.size() != 1)
+  {
+    success_msg_writer() << tr("usage: parse_address <addr>");
+    return true;
+  }
+
+  cryptonote::address_parse_info info;
+  if(get_account_address_from_str(info, m_wallet->testnet(), args.back()))
+  {
+    bool testnet = m_wallet->testnet();
+    success_msg_writer() << tr("Address string: ") << cryptonote::get_account_address_as_str(testnet, info.is_subaddress, info.address);
+    success_msg_writer() << tr("Type: ") << tr(info.is_subaddress ? "Subaddress" : "Standard address");
```

Comment:
Actually I was unaware of the purpose of `tr()` until recently and was using it without knowing what I'm doing :)